scripts/correct_video_timestamps.py

```
@@ -0,0 +1,271 @@
+        # Windows: use file creation time
+        ctime = file_path.stat().st_ctime
+        return datetime.fromtimestamp(ctime)
+    else:
+        # Unix: use modification time
+        mtime = file_path.stat().st_mtime
+        return datetime.fromtimestamp(mtime)
+
+
+def get_gpx_first_timestamp(gpx_file: str) -> datetime:
+    """Get first timestamp from GPX track."""
+    tree = ET.parse(gpx_file)
```

Ruff (0.15.14)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)